### PR TITLE
Nick/23

### PR DESCRIPTION
**Pull request by grittygrease**

### CODE REVIEW

File: draft-ietf-tls-exported-authenticator.md [hunk 1]
```diff
@@ -118,23 +118,32 @@ supporting certificates in the chain. This structure is defined in {{!TLS13}},
 Section 4.4.2.
 
 The certificate message contains an opaque string called
-certificate_request_context which SHOULD be unique for a given connection.  Its format
-is be defined by the application layer protocol and SHOULD be non-zero
-length.  For example, it may be a sequence number used by the higher-level
+certificate_request_context which MAY be set by the application to differentiate
```
Comment (martinthomson) [inline draft-ietf-tls-exported-authenticator.md]: s/MAY/can

File: draft-ietf-tls-exported-authenticator.md [hunk 2]
```diff
@@ -118,23 +118,32 @@ supporting certificates in the chain. This structure is defined in {{!TLS13}},
...
+exported authentictors.  Its format is be defined by the application layer protocol.
```
Comment (martinthomson) [inline draft-ietf-tls-exported-authenticator.md]: Ambiguous subject.  Try "The format of certificate_request_context".

File: draft-ietf-tls-exported-authenticator.md [hunk 3]
```diff
@@ -118,23 +118,32 @@ supporting certificates in the chain. This structure is defined in {{!TLS13}},
...
 an attacker with temporary access to a private key.
 
 CertificateVerify
-: A signature over the value
-Hash(Handshake Context || Certificate)
-
-This is described in Section 4.2.3 of {{!TLS13}}.  The signature scheme
-MUST be a valid signature scheme for TLS 1.3.  This excludes all RSASSA-PKCS1-v1_5
+: This message is used to provide explicit proof that an endpoint possesses the private key corresponding to its certificate.
+       struct {
```
Comment (martinthomson) [inline draft-ietf-tls-exported-authenticator.md]: Add a line break here, or the struct will be jammed onto the end of the sentence (which might be the source of your problems).

File: draft-ietf-tls-exported-authenticator.md [hunk 4]
```diff
@@ -118,23 +118,32 @@ supporting certificates in the chain. This structure is defined in {{!TLS13}},
...
+          SignatureScheme algorithm;
+          opaque signature<0..2^16-1>;
+       } CertificateVerify;
+
+The algorithm field specifies the signature algorithm used (see section 4.2.3 of {{!TLS13}}
```
Comment (martinthomson) [inline draft-ietf-tls-exported-authenticator.md]: s/section/Section

File: draft-ietf-tls-exported-authenticator.md [hunk 5]
```diff
@@ -118,23 +118,32 @@ supporting certificates in the chain. This structure is defined in {{!TLS13}},
...
+for the definition of this field). The signature is a digital signature using that algorithm.
+The signature scheme MUST be a valid signature scheme for TLS 1.3.  This excludes all RSASSA-PKCS1-v1_5
 algorithms and ECDSA algorithms that are not supported in TLS 1.3.  For servers,
 this signature scheme must match one of the signature and hash algorithms advertised
-in the signature_algorithms extension of the ClientHello.
+in the signature_algorithms extension of the ClientHello.  The signature is computed using the over the concatenation of:
+
+* A string that consists of octet 32 (0x20) repeated 64 times
+* The context string "version 1, Exported Authenticator"
```
Comment (martinthomson) [inline draft-ietf-tls-exported-authenticator.md]: Where does the "version 1" part come from?  You don't need to do that provided that there is no collision between the context string used here and the one used in TLS.  (Version 2 can add "version 2" or just change the name if we ever get there)

**Review by martinthomson — APPROVED** [review]
This looks generally right.

File: draft-ietf-tls-exported-authenticator.md [hunk 6]
```diff
@@ -118,23 +118,34 @@ supporting certificates in the chain. This structure is defined in {{!TLS13}},
...
+The algorithm field specifies the signature algorithm used (see Section 4.2.3 of {{!TLS13}}
+for the definition of this field). The signature is a digital signature using that algorithm.
+The signature scheme MUST be a valid signature scheme for TLS 1.3.  This excludes all RSASSA-PKCS1-v1_5
 algorithms and ECDSA algorithms that are not supported in TLS 1.3.  For servers,
 this signature scheme must match one of the signature and hash algorithms advertised
-in the signature_algorithms extension of the ClientHello.
+in the signature_algorithms extension of the ClientHello.  The signature is computed using the over the concatenation of:
+
+* A string that consists of octet 32 (0x20) repeated 64 times
+* The context string "Exported Authenticator"
```
Comment (wbl) [inline draft-ietf-tls-exported-authenticator.md]: Is this string itself null terminated? I think not, but it would be good to say so explicitly.

Comment (grittygrease) [reply draft-ietf-tls-exported-authenticator.md]: Good call, making the change.

**Pull request merged**